tensor_genn/algorithms/relu_ann.py

```diff
@@ -87,29 +96,30 @@ def convert(self, tf_model):
 
         return self.g_model
 
-    def evaluate(self, X, y=None, single_example_time=350.):
+    def evaluate(self, X, y=None):
         n_examples = len(X)
         X = X.reshape(n_examples,-1)
         y = y.reshape(n_examples)
 
-        runtime = n_examples * single_example_time
+        runtime = n_examples * self.single_example_time
+        n = len(self.neuron_pops)
         i = -1
         n_correct = 0
 
         while self.g_model.t < runtime:
-            if self.g_model.t >= single_example_time*(i+1):
+            if self.g_model.t >= self.single_example_time*(i+1):
                 # After example i -1,0,1,2,..
-                self.g_model.pull_var_from_device("if2",'SpikeNumber')
-                SpikeNumber_view = self.neuron_pops["if2"].vars["SpikeNumber"].view
+                self.g_model.pull_var_from_device("if"+str(n-1),'SpikeNumber')
+                SpikeNumber_view = self.neuron_pops[-1].vars["SpikeNumber"].view
                 n_correct += (np.argmax(SpikeNumber_view)==y[i])
                 i += 1
                 # Before example i 0,1,2,3,..
                 for j in range(len(self.neuron_pops)):
```

Comment:
Here you could do ``for j, n in enumerate(self.neuron_pops):`` and save some typing